nilmtk/dataset_converters/dred/convert_dred.py

```diff
@@ -0,0 +1,137 @@
+'''
+DRED Dataset converter.
+The .h5 format is hosted in DRED official website. But the file is not fully compatible with NILMTK.
+Download the .h5 file directly or download .csv and convert to .h5 using this converter.
+
+https://drive.google.com/open?id=1NDiRGVb33SQaKL_W7Y6WXEs9_xR1xTij
+
+CSV file name :- csvmerged.csv
+.h5 file name :- book1DRED.h5
+
+'''
+
+from __future__ import print_function, division
+import pandas as pd
+import numpy as np
+from copy import deepcopy
+from os.path import join, isdir, isfile
+from os import listdir
+import fnmatch
+import re
+from sys import stdout
+from nilmtk.utils import get_datastore
+from nilmtk.datastore import Key
+from nilmtk.timeframe import TimeFrame
+from nilmtk.measurement import LEVEL_NAMES
+from nilmtk.utils import get_module_directory, check_directory_exists
+from nilm_metadata import convert_yaml_to_hdf5, save_yaml_to_datastore
+
+
+def convert_dred(input_path, output_filename, format='HDF'):
+    """
+    Parameters
+    ----------
+    input_path : str
+        The root path of the CSV files, e.g. House1.csv
+    output_filename : str
+        The destination filename (including path and suffix).
+    format : str
+        format of output. Either 'HDF' or 'CSV'. Defaults to 'HDF'
+    """
+        
+    # Open DataStore
+    store = get_datastore(output_filename, format, mode='w')
+
+    # Convert raw data to DataStore
+    _convert(input_path, store, 'Europe/Amsterdam')
+
+    # Add metadata
+    save_yaml_to_datastore(join(get_module_directory(), 
+                              'dataset_converters', 
+                              'dred', 
+                              'metadata'),
+                         store)
+    store.close()
+
+    print("Done converting DRED to HDF5!")
+
+def _convert(input_path, store, tz, sort_index=True):
+    """
+    Parameters
+    ----------
+    input_path : str
+        The root path of the DRED dataset.
+    store : DataStore
+        The NILMTK DataStore object.
+    measurement_mapping_func : function
+        Must take these parameters:
+            - house_id
+            - chan_id
+        Function should return a list of tuples e.g. [('power', 'apparent')]
+    tz : str 
+        Timezone e.g. 'Europe/Amsterdam'
+    sort_index : bool
+    """
+
+    check_directory_exists(input_path)
+
+    # Iterate though all houses and channels
+    # house 14 is missing!
```

Comment:
why do we have house 14? I think this dataset has a single home? Please remove the lines no relevant to the converter